packages/flutter_tools/lib/src/build_system/filestore.dart

```diff
@@ -61,31 +61,36 @@ class FileHash {
   }
 }
 
-/// A globally accessible cache of file hashes.
+/// A globally accessible cache that determines if files are up to date.
 ///
 /// In cases where multiple targets read the same source files as inputs, we
-/// avoid recomputing or storing multiple copies of hashes by delegating
-/// through this class. All file hashes are held in memory during a build
-/// operation, and persisted to cache in the root build directory.
+/// avoid recomputing or storing multiple copies of stamps by delegating
+/// through this class. All file stamps are held in memory during a build
+/// operation, and persisted to cache in the build directory.
+///
+/// By default, md5 hashes are used as the file stamp. [_timestampModeEnabled]
+/// can be used to configure faster checks for incremental builds.
 ///
 /// The format of the file store is subject to change and not part of its API.
-class FileHashStore {
-  FileHashStore(this.environment, this.fileSystem) :
+class FileStore {
```

Comment:
How about a `FileStore` base class with two implementations `HashFileStore` and `TimestampFileStore`.

Reply:
Done